test/dotnet-compile-fsc.Tests/FSharpCompilerTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using FluentAssertions;
+using Xunit;
+
+namespace Microsoft.DotNet.Tools.Compiler.Tests
+{
+    public class FsharpCompilerTests
+    {
+        private readonly static string s_testProjectsRoot = Path.Combine(
+            AppContext.BaseDirectory, 
+            "TestAssets", 
+            "TestProjects", 
+            "FSharpTestProjects");
+
+        [Fact]
+        public void Compilation_of_CompileFailApp_Program_Fails()
+        {
+            var testProject = Path.Combine(s_testProjectsRoot, "CompileFailApp", "project.json"); 
+            var buildCommand = new BuildCommand(testProject);
+
+            var oldDirectory = Directory.GetCurrentDirectory();
+            Directory.SetCurrentDirectory(Path.GetDirectoryName(testProject));
+
+            buildCommand.Execute().Should().Fail();
+
+            Directory.SetCurrentDirectory(oldDirectory);
+        }
+
+        [Fact]
+        public void Compilation_of_TestAppWithArgs_Succeeds()
```

Comment:
I think I prefer to have the expectation at the end of the name, like this
`Compilation_of_valid_app_should_succeed`